### ISSUE TITLE
[Bug] Invisible collision ghost on hinge [20w09a]

### ISSUE THREAD

**Issue by Dwarf-LordPangolin** [tsunamayo/Starship-EVO#1637]
### Short Description of the issue:
This one so weird that I can't describe it, so I'll just describe in detail how to duplicate it.

### Steps to reproduce the issue:
1. Place a hinge sideways, like this:
![20200301175732_1](https://user-images.githubusercontent.com/33041786/75636766-313ee580-5be7-11ea-926c-728de4e481c8.jpg)

2. Attach blocks to the **moving part** (may be tricky due to [this bug](https://github.com/tsunamayo/Skywanderers/issues/1617)) to form a framework that extends away from the hinge like this:
![20200301175929_1](https://user-images.githubusercontent.com/33041786/75636721-c8f00400-5be6-11ea-80cc-c0cfbb94fd7a.jpg)

3. Delete the framework except for the orange blocks, then test to demonstrate that the hinge can move freely:
![20200301175950_1](https://user-images.githubusercontent.com/33041786/75636731-e1f8b500-5be6-11ea-8eee-b92aa47041cd.jpg)

4. Attach pipes (this seems to happen primarily with round blocks like pipes or guns) to the orange blocks, then check again that this can move freely:
![20200301180000_1](https://user-images.githubusercontent.com/33041786/75636772-3e5bd480-5be7-11ea-98c5-1d242b4476e7.jpg)

5. Attach another set of pipes to the first set of pipes. Don't move the hinge yet:
![20200301180007_1](https://user-images.githubusercontent.com/33041786/75636784-60555700-5be7-11ea-9938-aa9b9c514989.jpg)

6. Delete the second set of pipes, then check to see if the hinge can move freely. It should get hung up on something invisible at this point.
![20200301180014_1](https://user-images.githubusercontent.com/33041786/75636812-8d096e80-5be7-11ea-85e8-b6ffe2c14032.jpg)

### Environment
- Skywanderers version: 20w09a
- Graphics Card: NVIDIA GeForce GTX 1060 6GB

### Additional information:
[output_log.txt](https://github.com/tsunamayo/Skywanderers/files/4272811/output_log.txt)

**Comment by tsunamayo** [comment]
Whoa! Thanks quite a test case! I keep you posted ;)

**Comment by tsunamayo** [comment]
By the way as it is an entity collision issue you can debug Children Collision Point instead of the parent one. There is two separate logic for entity/entity and children entity collisions, children being more precise but more costly. Cheers

**Comment by Dwarf-LordPangolin** [comment]
I think I've found part of what's going on here! I was trying to angle some displays with hinges, and the small hinges couldn't move because of how I'd placed the displays. I deleted them, but the hinge kept getting hung up again on invisible obstacles. I turned on View Collision Points, and saw that the place it was getting stuck matched points that were floating in midair, unattached to anything.

However, those points were where the corners of the display _had_ been before I deleted it. Saving and reloading caused the invisible points to disappear, and the hinge to move properly. In other words, the thing that was blocking the hinge from moving was a brick that had already been deleted.

So it looks like part of this issue is that hinges aren't updating the collision points when attached things are deleted, but on save/reload instead.
